clientv3/integration/client_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"bytes"
+	"io/ioutil"
+	"path"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/coreos/etcd/clientv3"
+	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
+	"github.com/coreos/etcd/integration"
+	"github.com/coreos/etcd/lease"
+	"github.com/coreos/etcd/pkg/testutil"
+	"github.com/coreos/etcd/storage/backend"
+)
+
+func NewTestBackend(t *testing.T) (string, backend.Backend) {
+	tmpPath, err := ioutil.TempDir("", "lease")
+	if err != nil {
+		t.Fatalf("failed to create tmpdir (%v)", err)
+	}
+
+	return tmpPath, backend.New(path.Join(tmpPath, "be"), time.Second, 10000)
+}
+
+func TestKVPut(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	tests := []struct {
+		key, val string
+		ttl      int64
+	}{
+		{"foo", "bar", 0},
+		{"foo", "bar", 5},
+	}
+
+	for i, tt := range tests {
+		clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
+
+		kv := clientv3.NewKV(clus.RandClient())
+
+		leaseID := lease.NoLease
+		if tt.ttl > 0 {
+			lresp, err := clus.RandClient().Lease.LeaseCreate(
+				context.TODO(),
+				&pb.LeaseCreateRequest{TTL: tt.ttl})
+			if err != nil {
+				t.Fatalf("#%d: couldn't lease (%v)", i, err)
+			}
+			leaseID = lease.LeaseID(lresp.ID)
```

Comment:
@xiang90 Please review these lines. I am not sure what's the best way to test put with lease id.

I first tried

```go
if tt.ttl > 0 {
	leaseID = lease.LeaseID(i + 1)

	dir, be := NewTestBackend(t)
	defer os.RemoveAll(dir)
	defer be.Close()

	le := lease.NewLessor(be)

	if _, err := le.Grant(leaseID, tt.ttl); err != nil {
		t.Fatalf("#%d: couldn't grant lease (%v)", i, err)
	}	
}
```

But I was getting `lease not found` errors when I try to put.

Thanks